cmake/arch_build.cmake

```diff
@@ -339,6 +339,20 @@ function(process_arch SYSVAR)
   include_directories(${MISSION_SOURCE_DIR}/cfe/fsw/cfe-core/src/inc)
   include_directories(${MISSION_SOURCE_DIR}/cfe/cmake/target/inc)
     
+  # propagate any OSAL interface compile definitions and include directories to this build
+  # This is set as a directory property here at the top level so it will apply to all code.
+  # This includes MODULE libraries that do not directly/statically link with OSAL but still
+  # should be compiled with these flags.
+  get_target_property(OSAL_COMPILE_DEFINITIONS osal INTERFACE_COMPILE_DEFINITIONS)
```

Comment:
I am not sure if I understand the intention of this right but it looks like you could simply control this by specifying in the `osal` target's CMake file:

```
target_include_directories(osal
  PUBLIC
    ...
  INTERFACE
    ...
  PRIVATE
    ...
)
```

and the same with `target_compile_definitions`.

Reply:
`target_*` with `PUBLIC` and `INTERFACE` specifiers are designed to propagate corresponding definitions to their user targets.

Reply:
That's exactly what the OSAL CMakeLists.txt file is doing in the upcoming update.  However, it only works "implicitly" for targets which directly link with OSAL. For shared library/module targets that do not directly link with OSAL but yet still need to use the headers and a compatible set of definitions, this is why it needs to be done explicitly.

Unless there is another way of doing this for MODULE libraries?

Reply:
After your explanation I understand it better now.

There are also interface libraries: you could collect the needed flags and definitions of osal and attach them to an interface library `osal-interface`. The library/module targets could then link to the interface library instead of linking to `osal` directly.

But I am not sure if this will simplify existing setup.

https://cmake.org/cmake/help/v3.0/command/add_library.html (`INTERFACE`)

Reply:
Interface libraries are an interesting option... this might actually be a possibility for simplifying the way OSAL manages the flags internally, then we can just put this inside add_cfe_app().  Will not have time to revisit this in the near term however, so hopefully this approach is good enough for now.

Reply:
See #626 to track